RepClient/src/app/company/company.page.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { IonicModule } from '@ionic/angular';
+import { provideMockStore } from '@ngrx/store/testing';
+
+import { CompanyPage } from './company.page';
+
+describe('CompanyPage', () => {
+  let component: CompanyPage;
+  let fixture: ComponentFixture<CompanyPage>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ CompanyPage ],
+      imports: [IonicModule.forRoot()],
+      providers: [
+        provideMockStore()
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(CompanyPage);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```

Comment:
Any reason you're not testing sortBy, itemLookup and cancelLookup?

Reply:
More functionality to be added later this was more of an optional thing. Currently the test code would be a duplication of the .ts code as it checks and displays dummy data

Reply:
> I see you've gone from a genetic list page which you can use for multiple reasons to a specific company page

It really depends on what you need it for. As an example one list is virtual and the other is a normal list. This on its own brings the issue that you will have 3 templates in one component that you will need to maintain. A virtual list is a list that un-loads any item to keep the memory usage on a mobile phone efficient. On the other hand a company wouldnt need a virutal list as it would be an over kill. Then on one you would need to hook it up with the search etc, that would need event binding / exposure.
TLDR; you should use generic components yes, but in this case it would require 10x times the work as what i was after was very simple. What should be a generic component and its planned to be is the loading part :)